Scripts/models/logistics.py

```diff
@@ -4,99 +4,100 @@
-        self.lc_size_factors = ddm_params.w5 * np.log(lc_sizes)
+        self.lc_size_factors = ddm_params.size_factor * np.log(lc_sizes)
         self.cost_matrix = cost_matrix
         self.lc_indices = lc_indices
         self.lc_sizes = lc_sizes
-        self.w1a = ddm_params.w1a
-        self.w1b = ddm_params.w1b
-        self.w2 = ddm_params.w2
-        self.w3 = ddm_params.w3
-        self.w4 = ddm_params.w4
-        self.w5 = ddm_params.w5
-        self.temperature = 5.0
-
-    def sigmoid(self, x: np.ndarray) -> np.ndarray:
-        """Compute sigmoid function."""
-        return 1 / (1 + np.exp(-x))
+        self.orig_lc_detour = ddm_params.orig_lc_detour
+        self.lc_dest_detour = ddm_params.lc_dest_detour
+        self.constant_detour = ddm_params.constant_detour
+        self.orig_dest_direct = ddm_params.orig_dest_direct
+        self.constant_direct = ddm_params.constant_direct
+        self.size_factor = ddm_params.size_factor
+        self.scale = 5.0
 
     def softmax(self, x: np.ndarray, axis: int = -1) -> np.ndarray:
         """Compute softmax values."""
-        exp_x = np.exp(x - np.max(x, axis=axis, keepdims=True))
+        exp_x = np.exp(x)
         return exp_x / np.sum(exp_x, axis=axis, keepdims=True)
 
     def logsumexp(self, x: np.ndarray, axis: int = -1) -> np.ndarray:
         """Compute log sum exp."""
         max_x = np.max(x, axis=axis, keepdims=True)
         return np.log(np.sum(np.exp(x - max_x), axis=axis)) + np.squeeze(max_x)
 
-    def compute_weighted(self, origin_indices: Union[Sequence[int], np.ndarray] = None, 
+    def compute_utilities(self, origin_indices: Union[Sequence[int], np.ndarray] = None, 
                         destination_indices: Union[Sequence[int], np.ndarray] = None) -> np.ndarray:
-        """Compute weighted detour and direct costs for given origins and destinations."""
+        """Compute utilities for detour and direct routes for given origins and destinations."""
         k = self.lc_indices.shape[0]
         
         # Expand dimensions for broadcasting
         o_idx_exp = np.expand_dims(origin_indices, axis=1) # Shape: (n, 1)
         d_idx_exp = np.expand_dims(destination_indices, axis=1) # Shape: (n, 1)
         lc_exp = np.broadcast_to(self.lc_indices, (len(origin_indices), k)) # Shape: (n, k)
         
-        # weighted costs for k logistics centers
-        detour_size_factor = (self.sigmoid(self.w1a) * self.cost_matrix[o_idx_exp, lc_exp] +
-                 self.sigmoid(self.w1b) * self.cost_matrix[lc_exp, d_idx_exp] +
-                 self.w2) + self.lc_size_factors # Shape: (n, k)
+        # utilities for routes through k logistics centers
+        detour_utilities = (self.orig_lc_detour * self.cost_matrix[o_idx_exp, lc_exp] +
+                 self.lc_dest_detour * self.cost_matrix[lc_exp, d_idx_exp] +
+                 self.constant_detour) - self.lc_size_factors # Shape: (n, k)

```

Comment:
The sign of `self.lc_size_factors` seems to have been fixed doubly (i.e., it is still wrong).